src/storage/mvcc/reader/reader.rs

```diff
@@ -278,15 +284,117 @@ impl<S: EngineSnapshot> MvccReader<S> {
         Ok(())
     }
 
-    pub fn load_in_memory_pessimisitic_lock_range<F>(
+    fn next_pair_from_storage<F>(
+        &mut self,
+        start_key: Option<&Key>,
+        end_key: Option<&Key>,
+        filter: F,
+    ) -> Result<Option<(Key, Lock)>>
+    where
+        F: Fn(&Key, TxnLock<'_>) -> bool,
+    {
+        self.create_lock_cursor()?;
+        let cursor = self.lock_cursor.as_mut().unwrap();
+        if !self.lock_cursor_seeked {
```

Comment:
I don't suggest introducing such a new state to reader actually. The caller must know how the first call to this function determines the reader's state, and then the reader cannot be used to perform another `scan_lock` any more. Actually we don't like the fact for a long time that cursors in the reader are already stateful stuff.

For our current need, I think it would be better to work with cursors directly in the function `scan_locks`, instead of splitting to another function, and using `lock_cursor.next` directly when iterating.